rdrf/rdrf/views/registration_rdrf.py

```diff
@@ -111,19 +111,36 @@ def dispatch(self, request, *args, **kwargs):
 
 class EmbeddedRegistrationView(RdrfRegistrationView):
 
+    def __init__(self):
+        self.activate_language = None
+        self.language_code = None
+
     @csp_update(SCRIPT_SRC=registration_csp_script_src,
                 FRAME_SRC=registration_csp_frame_src,
                 FRAME_ANCESTORS=settings.EMBED_FRAME_ANCESTORS)
     @csrf_exempt
     def dispatch(self, request, *args, **kwargs):
-        language = request.GET.get('language', None)
 
-        if language:
+        # Activate a different language if the language URL parameter has been passed in
+        self.activate_language = 'language' in request.GET
+
+        if self.activate_language:
+            self.language_code = request.GET.get('language')
+
             from django.utils import translation
-            translation.activate(language)
+            translation.activate(self.language_code or settings.LANGUAGE_CODE)
 
         return super().dispatch(request, *args, **kwargs)
 
+    def render_to_response(self, context, **response_kwargs):
+        response = super().render_to_response(context, **response_kwargs)
+
+        # Persist language preference
+        if self.activate_language:
+            response.set_cookie(settings.LANGUAGE_COOKIE_NAME, self.language_code)
```

Comment:
Sorry thought I submitted this but it was still pending!

Can this be done in dispatch? Something like this:
```
request.session[settings.LANGUAGE_COOKIE_NAME] = language
```

Reply:
I can try.

Reply:
Doesn't work for me

There used to be a session variable for setting the language, but it's deprecated in Django 3, removed in Django 4. 
https://github.com/django/django/commit/a8e2a9bac6e548d6ab2e13af6171d2fdd3b8055b

Also, the django docs have a list of series of locations to determine the user's language preference, and the session is not in any of them
https://docs.djangoproject.com/en/3.0/topics/i18n/translation/#how-django-discovers-language-preference

Reply:
:+1: I think overriding `render_to_response` is unnecessary, so `response.set_cookie(...)` can be done in dispatch:

```
    def dispatch(self, request, *args, **kwargs):
        response = super().dispatch(request, *args, **kwargs)

        if language := request.GET.get('language'):
            translation.activate(language)
            response.set_cookie(settings.LANGUAGE_COOKIE_NAME, language)

        return response
```